errors.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+// type ErosError interface {
+//  Error() string
+//  ErrorCode() int
+// }
+
+type ErosError interface {
+    Error() string
+    Code() int
+}
+
+type ErosErrorImpl struct {
```

Comment:
Can't agree here. The only reason there even is `ErosErrorImpl` here is that Go make sit impossible to create an anonymous struct that implements an interface.

As it is, I think the name `ErosErrorImpl` makes it reasonably clear that its purpose is to be the one and only implementer of the interface. If that ever changes (quite unlikely IMO), we can think about changing the name to distinguish it from other implementations.